set.mm

```diff
@@ -730651,6 +730957,1157 @@ last digit of the integer part (for ` b = ` 10 the first digit before
+      impancom ) NCUBUCZFUDOZGBOZCUEOZUFZGCDPZHUGZGAPZCQZHBUHZQZUIYBYFYLYBYFRZY
+      GYJHEUJSZUHZYKEULOZYMGYNOZCNUKUMPZOZYGYOQEFUNIUOZYFYQYBYCYDYQYEYCYDYQYCBY
+      NGYCYNUEFUPPZBYCYNEFYCUQIYNURZUSJUTVAVBVCVDYMCVEOZYBRYSYMYBUUCYFUUCYBYEYC
+      UUCYDCVFVOVGVHCVIVJYNACDULEGHUUBLKVKVLYMYJHYNBYFYNBQYBYFYNUUABYFYNEFYCYDY
+      EVMIUUBUSJVNVDVPVQVRYFYBVSZYLYFUUDRZYGTYKUUEDVTYNYRWHQYQYSRVSYGTQMUAYNYRY
+      HMUGAPUAUGQZHYNUHZDYPDMUAYNYRUUGWAQYTMYNADULEUAHUUBLKWBWCUUFHYNEUJWDWEWFU
+      UEYSYQYFUUDYSVSYFYSYBYFYSUUCYBCUKUBUCZUFZYBNVEOZUKVEOZRYSUUIWGYFUUJUUKWIW
+      JWKNUKCWLWTUUCYBUUHWMWNWOWPWQGCYNYRDWRWSUUEYJVSZHBXAZTYKQZUUEUULHBUUEYHBO
+      ZUULYFUUOUUDUULYFUUORZYJYBUUPNYIUBUCZYJYBUUPABXBSZOUUOYDUUQUUPAUUAXBSZUUR
+      YFAUUSOZUUOYCYDUUTYEAFAEXCSFUNSZXCSKEUVAXCIXDXEXFXGXHBUUAXBJXDXIYFUUOXKYF
+      YDUUOYCYDYEWMXHYHGABXJXLYICNUBXMXNWOYAWPXOUUNYKTQUUMTYKXPYJHBXQXRVJVQXSXT
+      $.
+  $}
+
+  ${
+    $d E p $.  $d I p $.  $d M p $.  $d P p $.  $d R p $.
+    2sphere.i $e |- I = { 1 , 2 } $.
+    2sphere.e $e |- E = ( RR^ ` I ) $.
+    2sphere.p $e |- P = ( RR ^m I ) $.
+    2sphere.s $e |- S = ( Sphere ` E ) $.
+    ${
+      2sphere.c $e |- C = { p e. P | ( ( ( ( p ` 1 ) - ( M ` 1 ) ) ^ 2 )
+                         + ( ( ( p ` 2 ) - ( M ` 2 ) ) ^ 2 ) ) = ( R ^ 2 ) } $.
+      $( The sphere with center ` M ` and radius ` R ` in a two dimensional
+         Euclidean space is a circle.  (Contributed by AV, 5-Feb-2023.) $)
+      2sphere $p |- ( ( M e. P /\ R e. ( 0 [,) +oo ) ) -> ( M S R ) = C ) $=
+        ( wcel co wa cfv wceq cr c2 cc0 cpnf cico cds crab cfn cpr prfi eqeltri
+        cv c1 simpl cle elrege0 simplbi adantl eqid rrxsphere mp3an2i cmin cexp
+        wbr caddc biimpi ad2antlr sqrtsq syl eqeq2d wb rrx2pxel adantr resubcld
+        csqrt resqcld rrx2pyel readdcld sqge0d addge0d jca adantlr resqcl sqge0
+        sqrt11 syl2anc anim1ci crrx cehl 2nn0 ehlval ax-mp fz12pr eqtr4i fveq2i
+        cfz cn0 cmap oveq2i ehl2eudisval eqcomd eqeq1d 3bitr3d rabbidva syl5req
+        eqtri eqtrd ) GBNZCUAUBUCONZPZGCDOZHUJZGEUDQZOZCRZHBUEZAFUFNXHXFCSNZXIX
+        NRFUKTUGZUFIUKTUHUIXFXGULZXGXOXFXGXOUACUMVBZCUNZUOZUPXKBCDEFGHJKXKUQZLU
+        RUSXHAUKXJQZUKGQZUTOZTVAOZTXJQZTGQZUTOZTVAOZVCOZCTVAOZRZHBUEXNMXHYLXMHB
+        XHXJBNZPZYJVMQZYKVMQZRZYOCRYLXMYNYPCYOYNXOXRPZYPCRXGYRXFYMXGYRXSVDVECVF
+        VGVHYNYJSNZUAYJUMVBZPZYKSNZUAYKUMVBZPZYQYLVIXFYMUUAXGXFYMPZYSYTUUEYEYIU
+        UEYDUUEYBYCYMYBSNXFBFXJIKVJUPXFYCSNYMBFGIKVJVKVLZVNZUUEYHUUEYFYGYMYFSNX
+        FBFXJIKVOUPXFYGSNYMBFGIKVOVKVLZVNZVPUUEYEYIUUGUUIUUEYDUUFVQUUEYHUUHVQVR
+        VSVTXGUUDXFYMXGXOUUDXTXOUUBUUCCWACWBVSVGVEYJYKWCWDYNYOXLCYNXLYOYNYMXFPX
+        LYORXHXFYMXQWEXKEXJGBEFWFQZTWGQZJUUKUKTWNOZWFQZUUJTWONUUKUUMRWHUUKTUUKU
+        QWIWJUULFWFUULXPFWKIWLWMXDWLBSFWPOSXPWPOKFXPSWPIWQXDYAWRVGWSWTXAXBXCXE
+        $.
+    $}
+
+    ${
+      $d .0. p $.
+      2sphere0.0 $e |- .0. = ( I X. { 0 } ) $.
```

Comment:
I stopped on this formulation for a second, it is not very usual.
I was about to suggest `<" 0 0 ">`, but then I realized that indices do not match (0-based vs. 1-based). 
Maybe a comment would help?
Like:
> `( I X. { 0 } )` represents the constant function equal to `0` for both indice values `1` and `2`, i.e. the origin.

Reply:
I proved ~rrx0 (saying that `( I X. { 0 } )` is the zero ("origin") in a generalized real Euclidean space) and referenced it in the comment of ~ 2sphere0. I hope this is sufficient.

Reply:
Yes, perfect, thank you!